src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -1429,6 +1434,106 @@ CUDNN_SAFE_CALL(cudnnSetOpTensorDescriptor(opTensorDesc,
                 return;
             }
 
+            template <>
```

Comment:
Would it be possible to put some comments in here as to what the variable names are in the kernel? In the below kernel, am I correct assuming in assuming that `idx_in_start_1` is the index to the beginning of the window along axis 1? What is `idx_init`? Overall looks good, but if you can add some comments to help us understand in the future, that would be great.

```
extern "C" __global__ void cuda_reduce_window_max_float_float_float3(float* in, float* out, int* input_strides, int* output_shape, int* reduce_window_shape, int* reduce_window_strides, size_t n)
{
    const int tid = blockIdx.x * blockDim.x + threadIdx.x;
    if (tid < n)
    {
        int idx_out = tid;
        int idx_init = 0;
        int idx_out_2 = idx_out % output_shape[2];
        int idx_in_start_2 = idx_out_2 * reduce_window_strides[2];
        int idx_in_end_2 = idx_in_start_2 + reduce_window_shape[2];
        idx_init += idx_in_start_2 * input_strides[2];
        idx_out /= output_shape[2];
        int idx_out_1 = idx_out % output_shape[1];
        int idx_in_start_1 = idx_out_1 * reduce_window_strides[1];
        int idx_in_end_1 = idx_in_start_1 + reduce_window_shape[1];
        idx_init += idx_in_start_1 * input_strides[1];
        idx_out /= output_shape[1];
        int idx_out_0 = idx_out % output_shape[0];
        int idx_in_start_0 = idx_out_0 * reduce_window_strides[0];
        int idx_in_end_0 = idx_in_start_0 + reduce_window_shape[0];
        idx_init += idx_in_start_0 * input_strides[0];
        idx_out /= output_shape[0];
        float result = in[idx_init];
        for(int i_0 = idx_in_start_0; i_0 < idx_in_end_0; i_0++)
        {
            for(int i_1 = idx_in_start_1; i_1 < idx_in_end_1; i_1++)
            {
                for(int i_2 = idx_in_start_2; i_2 < idx_in_end_2; i_2++)
                {
                    int idx_in = 0;
                    idx_in += i_0 * input_strides[0];
                    idx_in += i_1 * input_strides[1];
                    idx_in += i_2 * input_strides[2];
                    result = (idx_in == idx_init) ? result : max(result, in[idx_in]);
                }
            }
        }
        out[tid] = result;
    }
}
```

Reply:
add some comments and change names to more meaningful ones.